runtime/array.c

```diff
@@ -522,8 +523,8 @@ static value caml_array_gather(intnat num_arrays,
 
     /* Many caml_initialize in a row can create a lot of old-to-young
        refs.  Give the minor GC a chance to run if it needs to.
-       Run memprof callback for the major allocation. */
-    res = caml_check_urgent_gc_and_callbacks (res);
+       Run memprof callbacks for the major allocation. */
+    caml_process_pending_actions_with_root (res);
```

Comment:
Don't forget to assign `res`:
```suggestion
    res = caml_process_pending_actions_with_root (res);
```

Reply:
Well spotted, thanks! I think this would have been easier to catch after squashing the fixup commit, but of course I should not have made it in the first place.